[2024-02-23 15:41:38,388: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-02-23 15:41:38,390: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 15:41:38,400: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 15:41:38,405: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 15:41:38,405: INFO: common: created directory at: artifacts]
[2024-02-23 15:41:38,405: INFO: common: created directory at: artifacts/data_ingestion]
[2024-02-23 15:41:39,200: ERROR: main: HTTP Error 404: Not Found]
Traceback (most recent call last):
  File "f:\Generative AI course\Projects\Wine-Quality-ML Project\Complete-ML-Project_Honey-Production-Forecasting\main.py", line 9, in <module>
    data_ingestion.main()
  File "f:\generative ai course\projects\wine-quality-ml project\complete-ml-project_honey-production-forecasting\src\Project\pipeline\_01_data_ingestion.py", line 17, in main
    data_ingestion.download_file()
  File "f:\generative ai course\projects\wine-quality-ml project\complete-ml-project_honey-production-forecasting\src\Project\components\data_ingestion.py", line 18, in download_file
    filename, headers = request.urlretrieve(
                        ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\urllib\request.py", line 241, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
                            ^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\urllib\request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\urllib\request.py", line 525, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\urllib\request.py", line 634, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\urllib\request.py", line 563, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\urllib\request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
[2024-02-23 15:43:15,840: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-02-23 15:43:15,840: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 15:43:15,840: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 15:43:15,840: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 15:43:15,840: INFO: common: created directory at: artifacts]
[2024-02-23 15:43:15,840: INFO: common: created directory at: artifacts/data_ingestion]
[2024-02-23 15:43:17,340: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 9937
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "01d477e7a5640075ce82ef629c54ae925c7dc1fc0bdfbc2eca152a193f3138c7"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 2334:2E8671:27ACF12:29495B7:65D8AF05
Accept-Ranges: bytes
Date: Fri, 23 Feb 2024 14:43:17 GMT
Via: 1.1 varnish
X-Served-By: cache-fra-eddf8230109-FRA
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1708699398.544717,VS0,VE157
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 0316ba71f9dafe781f66c0b20f16cd82766a6dca
Expires: Fri, 23 Feb 2024 14:48:17 GMT
Source-Age: 0

]
[2024-02-23 15:43:17,378: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-02-23 16:00:51,016: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:00:51,518: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-02-23 16:00:51,522: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 16:00:51,525: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 16:00:51,527: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 16:00:51,527: INFO: common: created directory at: artifacts]
[2024-02-23 16:00:51,527: INFO: common: created directory at: artifacts/data_ingestion]
[2024-02-23 16:00:51,531: INFO: data_ingestion: File already exists of size: ~ 10 KB]
[2024-02-23 16:00:51,532: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-02-23 16:00:51,535: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2024-02-23 16:00:51,537: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 16:00:51,537: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 16:00:51,542: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 16:00:51,543: INFO: common: created directory at: artifacts]
[2024-02-23 16:00:51,544: INFO: common: created directory at: artifacts/data_validation]
[2024-02-23 16:00:51,575: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2024-02-23 16:02:55,783: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:02:56,178: INFO: _02_data_validation: >>>>>> stage Data Validation stage started <<<<<<]
[2024-02-23 16:02:56,181: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 16:02:56,183: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 16:02:56,185: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 16:02:56,186: INFO: common: created directory at: artifacts]
[2024-02-23 16:02:56,186: INFO: common: created directory at: artifacts/data_validation]
[2024-02-23 16:02:56,199: INFO: _02_data_validation: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2024-02-23 16:14:47,690: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:14:47,994: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:14:48,009: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:16:54,334: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:16:54,660: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:16:54,660: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:24:02,570: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:24:02,879: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:24:02,879: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:24:26,439: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:24:26,823: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:24:26,825: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:25:16,221: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:25:16,556: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:25:16,558: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:25:41,868: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:25:42,194: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:25:42,196: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:26:02,229: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:26:02,597: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:26:02,597: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:27:53,520: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:27:53,829: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:27:53,838: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:29:27,439: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:29:27,753: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:29:27,753: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:29:37,123: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:29:37,556: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:29:37,559: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:30:14,440: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:30:14,808: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:30:14,819: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:33:19,341: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:33:19,714: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:33:19,715: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:33:40,707: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:33:41,072: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:33:41,072: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:34:43,656: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:34:44,024: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:34:44,024: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:34:56,192: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:34:56,530: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:34:56,530: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:35:47,906: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:35:49,147: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:35:49,148: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:38:17,489: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:38:17,832: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:38:17,832: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:40:27,908: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:40:28,303: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-02-23 16:40:28,404: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 16:40:28,407: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 16:40:28,409: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 16:40:28,410: INFO: common: created directory at: artifacts]
[2024-02-23 16:40:28,411: INFO: common: created directory at: artifacts/data_transformation]
[2024-02-23 16:40:28,449: INFO: data_transformation: Splited data into training and test sets]
[2024-02-23 16:40:28,449: INFO: data_transformation: (469, 5)]
[2024-02-23 16:40:28,449: INFO: data_transformation: (157, 5)]
[2024-02-23 16:40:28,457: INFO: _03_data_transformation: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-02-23 16:52:29,179: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:52:30,453: INFO: _04_model_trainer: >>>>>> stage Model Trainer stage started <<<<<<]
[2024-02-23 16:52:30,456: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 16:52:30,456: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 16:52:30,456: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 16:52:30,456: INFO: common: created directory at: artifacts]
[2024-02-23 16:52:30,462: ERROR: _04_model_trainer: "'ConfigBox' object has no attribute 'elasticnet'"]
Traceback (most recent call last):
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 503, in __getitem__
    return super().__getitem__(item)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'ElasticNet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 536, in __getattr__
    value = self.__getitem__(item, _ignore_default=True)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 524, in __getitem__
    raise BoxKeyError(str(err)) from _exception_cause(err)
box.exceptions.BoxKeyError: "'ElasticNet'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 538, in __getattr__
    value = object.__getattribute__(self, item)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ConfigBox' object has no attribute 'ElasticNet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\config_box.py", line 28, in __getattr__
    return super().__getattr__(item)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 552, in __getattr__
    raise BoxKeyError(str(err)) from _exception_cause(err)
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'ElasticNet'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 503, in __getitem__
    return super().__getitem__(item)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'elasticnet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 536, in __getattr__
    value = self.__getitem__(item, _ignore_default=True)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 524, in __getitem__
    raise BoxKeyError(str(err)) from _exception_cause(err)
box.exceptions.BoxKeyError: "'elasticnet'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 538, in __getattr__
    value = object.__getattribute__(self, item)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ConfigBox' object has no attribute 'elasticnet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "f:\Generative AI course\Projects\Wine-Quality-ML Project\Complete-ML-Project_Honey-Production-Forecasting\src\Project\pipeline\_04_model_trainer.py", line 25, in <module>
    obj.main()
  File "f:\Generative AI course\Projects\Wine-Quality-ML Project\Complete-ML-Project_Honey-Production-Forecasting\src\Project\pipeline\_04_model_trainer.py", line 15, in main
    model_trainer_config = config.get_model_trainer_config()
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "f:\generative ai course\projects\wine-quality-ml project\complete-ml-project_honey-production-forecasting\src\Project\config\configuration.py", line 72, in get_model_trainer_config
    params = self.params.ElasticNet
             ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\config_box.py", line 30, in __getattr__
    return super().__getattr__(item.lower())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dell\anaconda3\envs\IPC_kernel\Lib\site-packages\box\box.py", line 552, in __getattr__
    raise BoxKeyError(str(err)) from _exception_cause(err)
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'elasticnet'"
[2024-02-23 16:53:23,080: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:53:24,284: INFO: _04_model_trainer: >>>>>> stage Model Trainer stage started <<<<<<]
[2024-02-23 16:53:24,284: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 16:53:24,284: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 16:53:24,284: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 16:53:24,284: INFO: common: created directory at: artifacts]
[2024-02-23 16:53:24,293: INFO: common: created directory at: artifacts/model_trainer]
[2024-02-23 16:53:24,322: INFO: _04_model_trainer: >>>>>> stage Model Trainer stage completed <<<<<<

x==========x]
[2024-02-23 16:58:51,906: INFO: utils: NumExpr defaulting to 4 threads.]
[2024-02-23 16:58:53,121: INFO: _05_model_evaluation: >>>>>> stage Model evaluation stage started <<<<<<]
[2024-02-23 16:58:53,121: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-02-23 16:58:53,121: INFO: common: yaml file: params.yaml loaded successfully]
[2024-02-23 16:58:53,121: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-02-23 16:58:53,136: INFO: common: created directory at: artifacts]
[2024-02-23 16:58:53,137: INFO: common: created directory at: artifacts/model_evaluation]
[2024-02-23 16:58:53,244: INFO: common: json file saved at: artifacts\model_evaluation\metrics.json]
[2024-02-23 16:58:53,245: INFO: _05_model_evaluation: >>>>>> stage Model evaluation stage completed <<<<<<

x==========x]
